form/src/main/java/com/ritense/form/domain/FormIoFormDefinition.java

```diff
@@ -408,9 +408,14 @@ private static List<ArrayNode> getComponents(JsonNode formDefinition) {
     }
 
     private Object extractNodeValue(JsonNode node) {
-        if (node.isValueNode()) {
+        var nodeType = node.getNodeType();
+        if (nodeType == JsonNodeType.STRING) {
             return node.textValue();
-        } else if (node.isArray()) {
+        } else if (nodeType == JsonNodeType.NUMBER) {
+            return node.numberValue();
```

Comment:
This is a bit of a breaking change. Processes that used to have a process variable with type 'string' can now get the same process variable with type 'boolean' or 'number'.

So this PR should go to major 10. We have a branch for that, can you change the destination to the major 10 branch?

Reply:
> This is a bit of a breaking change. Processes that used to have a process variable with type 'string' can now get the same process variable with type 'boolean' or 'number'.

Before this change this method would make every valueNode other than TextNode a null value. We always had the ability to submit what ever we wanted in form-io - that means that even before this fix we could've made a previously string value into a integer, thus the end result becoming null in the processvariable in camunda.
The only thing I now did is that the returned value aligns with it's nodeType - how is this breaking?

> So this PR should go to major 10. We have a branch for that, can you change the destination to the major 10 branch?

I need the valueNode types fixed asap, for containerNodes you can take this into major 10 if you like. If you don't want to accept this PR on the premise that it only fixes valueNodes in the extractNodeValue method, then close the PR and I'll make a story for team Valtimo instead.

Reply:
A right, values of other types become null